contracts/Strategy.sol

```diff
@@ -274,60 +371,317 @@ contract Strategy is BaseStrategy {
         (_liquidatedAmount, ) = liquidatePosition(type(uint256).max);
     }
 
-    // The _newStrategy must support the same kashiPair or bad things will happen
-    function prepareMigration(address _newStrategy) internal override {
-        kashiPair.transfer(_newStrategy, kashiFraction());
-    }
+    // We stake in the masterChef so we can't transfer assets when migrating
+    function prepareMigration(address _newStrategy) internal override {}
```

Comment:
couldn't you unstake from masterchef and transfer? 

looks a lot simpler than unwinding the whole position

Reply:
My hesitation is you will need to do something to ensure all the kashi tokens are staked after migration.

Reply:
yeah, you will need to harvest. 

I agree it is not 100% clear which is the best option. I don't have a strong view here

Reply:
Right now, calling harvest will not visit all kashiPairs and ensure kashi tokens are staked. This is to reduce the gas cost of a harvest. Given that, I think we should leave migrate unimplemented.

Reply:
gotcha

Reply:
Ended up taking your advice. Migration implemented and harvest will restate.